src/rust/engine/options/src/lib.rs

```diff
@@ -489,19 +489,30 @@ impl OptionParser {
             }
             derivation = Some(derivations);
         }
+
+        // Removals from any source apply after adds from any source (but are themselves
+        // overridden by later replacements), so we collect them here and apply them later.
+        let mut removal_lists: Vec<Vec<T>> = vec![];
```

Comment:
Totally minor detail (that doesn't need to change), but could this be `Vec<T>` with `extend` instead of `push` and then a single `remover` call.

Reply:
Yeah, I was 50-50 on which I preferred, kinda chose this arbitrarily.